### DISCUSSION TITLE
Multiple Databases for the same entity

### DISCUSSION THREAD

**Discussion by dimpdash** [mikro-orm/mikro-orm#3962] [Q&A]
Hoping to add some new drivers but want to make sure my goal is possible.

My team is currently using dynamoDB as a database but have larger data referenced inside of S3. DDB still has some of the entities data but one column may be flagged as being stored in S3. 

Would this require a drastic overhaul? I know current mikro does not have support for multiple databases.

My current plan was to build out the two DDB and S3 drivers then create a third driver that composes them together. The third driver would

- on retrieval check for the flag and call s3 if necessary
- on insertion split the data up

**Comment by B4nan** [comment]
> I know current mikro does not have support for multiple databases.

What do you mean by that? You can have many ORM instances, each using a different database. They can share the same entity definitions as long as they don't contain anything driver specific (you can move such things into platform level and use custom mapped types, so that can be gotten around as well).

> My current plan was to build out the two DDB and S3 drivers and then create a third driver that composes them together. The third driver would

If you want to mix this in a single ORM instance, you need a single driver - but inside its implementation, you can technically use multiple databases.

**Reply by dimpdash** [reply]
Thanks for the reply. I like the suggestion about two orms. 

As a callout for other readers. Moving forward, I'll try to build out the two base drivers then decide if two orm instances or a mix driver approach is the way to go.
